### ISSUE TITLE
question: span name change broke metrics?

### ISSUE THREAD

**Issue by dcormier** [DataDog/opencensus-go-exporter-datadog#43]
Prior to #36, setting a span name would do what one would expect: set the name of the span. PR #36 changed that so that _all span names are named `opencensus`_, and setting the span name actually sets the resource. Yes, you can override it _one span at a time_ by using the special `span.name` attribute, but that's impractical for those of us with large codebases with hundreds of spans where we were already setting the span name.

One of the most problematic effects of this is that now the automatic trace metrics DataDog has based on span name are all for the `opencensus` span (metrics named `trace.opencensus.*`). Which means there's now just one set of trace metrics. Not very useful.

**TL; DR:** Changing the span names to all be the same cause some things to break. It would be better if this were optional.

**Comment by gbbr** [comment]
You're right. I think we should revert this change. This change happened for situations when the name was invalid in the Datadog system. We should perhaps revert this change and make it so that the default `opencensus` name is only set when the name is invalid.

**Comment by gbbr** [comment]
If we do this, it will break everything reported in #35 again, and I tend to believe that's worse than the problem you are describing. Did you know that you can filter those metrics by tags `service`, `resource` and `resource_name`? Does that change anything for you?

**Comment by dcormier** [comment]
That does seem to work for what I'm currently trying to do.

**Comment by gbbr** [comment]
Great! Let's not make any changes then. Should we then close this issue?

**Comment by dcormier** [comment]
I think that's OK.

**Comment by gbbr** [comment]
Thanks Daniel. Glad it worked out for you.
